ts/util/dropdown.ts

```diff
@@ -0,0 +1,72 @@
+interface DropdownOption {
+  render(li: HTMLLIElement): void;
+}
+export class BasicDropdownOption implements DropdownOption {
+  text: string;
+  handler: () => void;
+
+  constructor(text: string, handler: () => void) {
+    this.text = text;
+    this.handler = handler;
+  }
+
+  render(li: HTMLLIElement): void {
+    li.className = "px-2 py-1 rounded-md hover:bg-green-200 hover:text-white";
+    li.innerHTML = this.text;
+    li.addEventListener("click", this.handler);
+  }
+}
+
+export function makeDropdownMenu(
+  DropdownOptions: BasicDropdownOption[],
+  defaultOption: string
+) {
+  const dropdownButton = makeDropdownButton();
+  dropdownButton.innerHTML = defaultOption;
+  const dropdownContainer = document.createElement("div");
+  const dropdownClickout = makeDropdownClickout();
+
+  const dropdownMenuContainer = makeDropdownMenuContainer();
+  DropdownOptions.forEach((option) => {
+    const li = document.createElement("li");
+    option.render(li);
+    li.addEventListener(
+      "click",
+      () => (dropdownButton.innerHTML = option.text)
```

Comment:
This should close the dropdown as well. It's currently staying open when to select an option